### PR TITLE
improve the compaction

### PR DESCRIPTION
**Pull request by git-hulk**
the main goal of the compaction checker was to avoid unnecessary full compaction every day. How it works:
* step 1: check the current hour whether in compaction checker range or not, if yes go to step 2
* step 2: fetch all table properties and then pick the file was created
more than 1 week or which has highest delete ratio every minute
* step 3: compact keys in the selected file

### TODO

- [x]  fix pubsub/slot compaction

- [x]  test compaction checker was worked as expected

- [ ] smarter compaction rule

### CODE REVIEW

File: src/storage.h [hunk 1]
```diff
@@ -70,6 +70,7 @@ class Storage {
   void IncrFlushCount(uint64_t n) { flush_count_.fetch_add(n); }
   uint64_t GetCompactionCount() { return compaction_count_; }
   void IncrCompactionCount(uint64_t n) { compaction_count_.fetch_add(n); }
+  bool CondisEnabled() { return config_->codis_enabled; }
```
Comment (karelrooted) [inline src/storage.h:73]: typo

File: src/storage.cc [hunk 2]
```diff
@@ -174,6 +178,8 @@ Status Storage::Open(bool read_only) {
   subkey_opts.table_factory.reset(rocksdb::NewBlockBasedTableFactory(subkey_table_opts));
   subkey_opts.compaction_filter_factory = std::make_shared<SubKeyFilterFactory>(this);
   subkey_opts.disable_auto_compactions = config_->RocksDB.disable_auto_compactions;
+  metadata_opts.table_properties_collector_factories.emplace_back(
```
Comment (karelrooted) [inline src/storage.cc:181]: typo, metadata_opts => subkey_opts

Comment (git-hulk) [reply src/storage.cc:181]: good catch

**Comment by karelrooted** [comment]
LGTM

**Pull request merged**